galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/routes/PropertyRoute.java

```diff
@@ -99,6 +118,25 @@ protected  Map<String, String> filterPropertiesBySuffix( Map<String, String> pro
         return filteredProperties;
     }
 
+    /**
+     * Filter a map of provided properties by checking that the properties contain and match at least of the
+     * supplied infixes 
+     * @param properties
+     * @param infixes
+     * @return Map of Properties containing the at least one of the infixes
+     */
+    protected  Map<String, String> filterPropertiesByInfix(Map<String, String> properties, List<String> infixes){
+        Map<String, String> filteredProperties = new HashMap<String,String>();
+        for (Map.Entry<String, String> entry : properties.entrySet()) {
+            for (String infix : infixes){
+				if (entry.getKey().toString().contains(infix)&& !filteredProperties.containsKey(entry.getKey())){
+                    filteredProperties.put(entry.getKey(), entry.getValue());
```

Comment:
Really minor but you could store the `entry.getKey()` result in a variable and use that instead of repeatedly calling `getKey()`

Also, you might be able to get away with not using the `.toString()` since the key should already be a String, same applies to the method above that isn't in your PR

Reply:
done